vmdb/app/models/ems_refresh/parsers/kubernetes.rb

```diff
@@ -138,23 +138,14 @@ def parse_endpoint(entity)
       new_result = parse_base_item(entity)
       new_result[:container_groups] = []
 
-      # TODO: remove once kubernetes api v1beta3/v1.0 will be stable
-      if !entity.respond_to?(:subsets) || entity.subsets.nil?
-        $log.warn("your kubernetes endpoints are not providing subsets, " \
-                  "please upgrade, see https://github.com/" \
-                  "GoogleCloudPlatform/kubernetes/pull/5939")
-        return new_result
-      end
-
-      entity.subsets.each do |subset|
-        subset.fetch('addresses', []).each do |address|
-          target_ref = address['targetRef']
-          next if target_ref.nil? || target_ref['kind'] != 'Pod'
-          cg = @data_index.fetch_path(
-            :container_groups, :by_namespace_and_name, target_ref['namespace'],
-            target_ref['name'])
-          new_result[:container_groups] << cg unless cg.nil?
-        end
+      (entity.endpoints || []).each do |endpoint|
+        next if endpoint.targetRef.nil? || endpoint.targetRef.kind != 'Pod'
```

Comment:
I think you can simplify this to: `next unless endpoint.targetRef.try(:kind) == 'Pod'`